File: android/src/main/java/com/capacitor/plugin/env/CapacitorEnvPlugin.java
package com.capacitor.plugin.env;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

import org.json.JSONObject;
import java.util.Iterator;

@CapacitorPlugin(name = "CapacitorEnv")
public class CapacitorEnvPlugin extends Plugin {

    private CapacitorEnv implementation = new CapacitorEnv();

    // @PluginMethod
    // public void echo(PluginCall call) {
    //     String value = call.getString("value");

    //     JSObject ret = new JSObject();
    //     ret.put("value", implementation.echo(value));
    //     call.resolve(ret);
    // }

    @PluginMethod
    public void get(PluginCall call) {
        try {
            call.success(getConfig().getConfigJSON());
        } catch (Exception e) {
            call.error("Error reading CapacitorEnv values: " + e.getMessage());
        }
    }
}

File: ios/Plugin/CapacitorEnvPlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(CapacitorEnvPlugin)
public class CapacitorEnvPlugin: CAPPlugin {
    private let implementation = CapacitorEnv()

    @objc func echo(_ call: CAPPluginCall) {
        let value = call.getString("value") ?? ""
        call.resolve([
            "value": implementation.echo(value)
        ])
    }
    
    @objc func get(_ call: CAPPluginCall) {
        let config = getConfig() as PluginConfig
        let json = config.getConfigJSON()
        call.resolve(json)
    }
}
